src/users.rs

```diff
@@ -52,12 +52,20 @@ pub fn get_users_list() -> Vec<User> {
                                     groups: groups[..current as usize]
                                         .iter()
                                         .filter_map(|id| {
-                                            let g = getgrgid(*id as _);
-                                            if g.is_null() {
+                                             let mut g: libc::group = std::mem::zeroed();
+                                             let mut tmp_ptr = std::ptr::null_mut();
+                                             let mut buf = vec![0; 2048];
+                                             if getgrgid_r(
```

Comment:
`getgrgid_r` can fail because of EINTR. You can use the `retry_eintr!` macro to prevent this issue:

```rust
if retry_eintr!(getgrgid_r(*id as _, &mut g as _, buf.as_mut_ptr(), buf.capacity() as _, &mut tmp_ptr as _) != 0 {
    return None;
}
```

Reply:
man, I really wish I knew of this macro before.  would've saved me some time.
Thanks.

Reply:
It's not an official macro or anything. Just copy-paste the code of it (it's in `src/macros.rs` iirc).

Reply:
ok, thanks